services/codeclimate/codeclimate-coverage.service.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+const Joi = require('joi')
+const { BaseJsonService, NotFound } = require('..')
+const { coveragePercentage, letterScore } = require('../color-formatters')
+const { keywords, fetchRepo } = require('./codeclimate-common')
+
+const schema = Joi.object({
+  data: Joi.object({
+    attributes: Joi.object({
+      covered_percent: Joi.number().required(),
+      rating: Joi.object({
+        letter: Joi.equal('A', 'B', 'C', 'D', 'E', 'F').required(),
+      }).required(),
+    }).required(),
+  }).allow(null),
+}).required()
+
+module.exports = class CodeclimateCoverage extends BaseJsonService {
+  static get route() {
+    return {
+      base: 'codeclimate',
+      pattern: ':which(coverage|coverage-letter)/:user/:repo',
+    }
+  }
+
+  static get category() {
+    return 'coverage'
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Code Climate coverage',
+        namedParams: { which: 'coverage', user: 'jekyll', repo: 'jekyll' },
+        staticPreview: this.render({
+          which: 'coverage',
+          percentage: 95.123,
+          letter: 'A',
+        }),
+        keywords,
+      },
+    ]
+  }
+
+  async fetch({ user, repo }) {
+    const {
+      id: repoId,
+      relationships: {
+        latest_default_branch_test_report: { data: testReportInfo },
+      },
+    } = await fetchRepo(this, { user, repo })
+    if (testReportInfo === null) {
+      throw new NotFound({ prettyMessage: 'test report not found' })
+    }
+    const { data } = await this._requestJson({
```

Comment:
I did not know we had to make two API calls to get this data from Code Climate. That's unfortunate they don't make it available in one pass

Reply:
Perhaps as a future enhancement we could consider allowing users to enter their repo id directly to avoid making that first API call (we could update the docs with steps on how users can find their respective repo id's)

I imagine that might help a bit with the rate limiting issue described in #1518

Reply:
That’s a nice idea! A decent caching layer (a la #2547) could also help with stuff that without making the interface more opaque.